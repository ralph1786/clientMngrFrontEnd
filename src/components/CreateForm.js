import React, { Component } from "react";
import { withRouter, Redirect } from "react-router-dom";
import { createChild } from "../actions/childActions";
import { connect } from "react-redux";
import "./CreateForm.scss";
import { toast } from "react-toastify";
import { setProvider } from "../actions/authActions";
import ParentSelect from "./ParentSelect";
import CancelButton from "./CancelButton";

class CreateForm extends Component {
  state = {
    name: "",
    image: "",
    age: "",
    address: "",
    allergies: "",
    balance: "",
    provider_id: 0,
    parent_id: 1,
    forms: null
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  componentDidMount() {
    let token = localStorage.token;
    if (token) {
      fetch("http://localhost:80/api/v1/dashboard", {
        method: "GET",
        headers: {
          "content-type": "application/json",
          accepts: "application/json",
          Authorization: `Bearer ${token}`
        }
      })
        .then(res => res.json())
        .then(res => {
          if (res.error) {
            return <Redirect to="/login" />;
          } else {
            this.props.setProvider(res.provider);
            this.setState({
              provider_id: res.provider.id
            });
          }
        })
        .catch(err => console.log(err));
    }
  }

  handleFilesChange = e => {
    this.setState({
      forms: e.target.files
    });
  };

  handleCreateSubmit = e => {
    e.preventDefault();
    const data = new FormData();
    for (let key in this.state) {
      if (key === "forms") {
        for (let form of this.state.forms) {
          data.append("forms[]", form);
        }
      } else {
        data.append(key, this.state[key]);
      }
    }
    fetch("http://localhost:80/api/v1/children", {
      method: "POST",
      body: data
    })
      .then(res => res.json())
      .then(res => {
        toast.success("Successfully created!", {
          position: toast.POSITION.BOTTOM_CENTER
        });
        this.props.history.push("/dashboard");
      })
      .catch(err => console.log(err));
  };

  render() {
    return (
      <div className="create-form animated flipInY">
        <form onSubmit={this.handleCreateSubmit}>
          <label>Name</label>
          <input
            name="name"
            type="text"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Image Url</label>
          <input
            name="image"
            type="text"
            value={this.state.image}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Age</label>
          <input
            name="age"
            type="text"
            value={this.state.age}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Address</label>
          <input
            name="address"
            type="text"
            value={this.state.address}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Allergies</label>
          <input
            name="allergies"
            type="text"
            value={this.state.allergies}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Balance</label>
          <input
            name="balance"
            type="text"
            value={this.state.balance}
            onChange={this.handleChange}
          />
          <br />
          <br />
          <label>Parent</label>
          <ParentSelect
            name="parent_id"
            value={this.state.parent_id}
            handleSelectChange={this.handleChange}
          />
          <br />
          <br />
          <input
            name="forms"
            type="file"
            multiple
            onChange={this.handleFilesChange}
          />
          <br />
          <br />
          <input
            style={{ outline: "none" }}
            type="submit"
            value="Create Child"
          />
          <CancelButton pathname={this.props.history.location.pathname} />
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    provider: state.provider
  };
};

const mapDispatchToProps = dispatch => {
  return {
    newChild: childInfo => dispatch(createChild(childInfo)),
    setProvider: providerObj => dispatch(setProvider(providerObj))
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(CreateForm)
);
