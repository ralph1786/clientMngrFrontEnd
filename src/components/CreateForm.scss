.create-form {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: repeat(12, 1fr);
}

.create-form form {
  -webkit-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  -moz-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  background-color: rgb(32, 221, 158);
  grid-column: 5/9;
  grid-row: 1/5;
  text-align: center;
  font-size: 1.8em;
  margin: 10px;
  padding: 10px;
}

.create-form form label,
.create-form form input {
  margin-top: 5px;
  padding: 5px;
  height: 15px;
}

.create-form form {
  label:last-of-type {
    margin-right: 40px;
  }
  select {
    margin-left: 10px;
    width: 45%;
  }
}
.create-form form input[type="text"],
.create-form form input[type="number"] {
  //   margin-left: 5px;
  width: 50%;
  font-size: 0.8em;
  outline: none;
  float: right;
  margin-right: 20px;
}

.create-form form label {
  color: rgb(80, 78, 78);
}

.create-form form input[type="submit"] {
  width: 50%;
  height: 10%;
  font-size: 20px;
  margin-top: 10px;
  margin-bottom: 10px;
  border-radius: 25px;
}

.create-form form input[type="submit"]:hover {
  cursor: pointer;
  background-color: rgb(209, 208, 208);
}

#button-create {
  margin-left: 50px;
  margin-top: 115px;
  padding: 9px;
  border-radius: 25px;
  font-size: 1.04em;
  background-color: rgb(41, 125, 237);
  color: white;
  outline: none;
}

#button-create:hover {
  cursor: pointer;
  background-color: rgb(83, 152, 241);
}
