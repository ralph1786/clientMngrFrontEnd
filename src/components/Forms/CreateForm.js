import React, { Component } from "react";
import { withRouter, Redirect } from "react-router-dom";
import { createChild } from "../../actions/childActions";
import { connect } from "react-redux";
import "./CreateForm.scss";
import { toast } from "react-toastify";
import { setProvider } from "../../actions/authActions";
import ParentSelect from "../ParentSelect";
import CancelButton from "../UI/CancelButton";
import LabelInput from "./LabelInput";

export class CreateForm extends Component {
  state = {
    name: "",
    image: "",
    age: "",
    address: "",
    allergies: "",
    balance: "",
    provider_id: 0,
    parent_id: 1,
    forms: null
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  componentDidMount() {
    let token = localStorage.token;
    if (token) {
      fetch("http://localhost:80/api/v1/dashboard", {
        method: "GET",
        headers: {
          "content-type": "application/json",
          accepts: "application/json",
          Authorization: `Bearer ${token}`
        }
      })
        .then(res => res.json())
        .then(res => {
          if (res.error) {
            return <Redirect to="/login" />;
          } else {
            this.props.setProvider(res.provider);
            this.setState({
              provider_id: res.provider.id
            });
          }
        })
        .catch(err => console.log(err));
    }
  }

  handleFilesChange = e => {
    this.setState({
      forms: e.target.files
    });
  };

  handleCreateSubmit = e => {
    e.preventDefault();
    if (
      Object.values(this.state).includes("") ||
      Object.values(this.state).includes(null)
    ) {
      toast.error("All Fields Required!", {
        position: toast.POSITION.TOP_CENTER,
        autoClose: 5000
      });
      return;
    }
    const data = new FormData();
    for (let key in this.state) {
      if (key === "forms") {
        for (let form of this.state.forms) {
          data.append("forms[]", form);
        }
      } else {
        data.append(key, this.state[key]);
      }
    }
    fetch("http://localhost:80/api/v1/children", {
      method: "POST",
      body: data
    })
      .then(res => res.json())
      .then(res => {
        toast.success("Successfully created!", {
          position: toast.POSITION.BOTTOM_CENTER
        });
        this.props.history.push("/dashboard");
      })
      .catch(err => console.log(err));
  };

  render() {
    return (
      <div className="create-form animated flipInY">
        <form onSubmit={this.handleCreateSubmit}>
          <LabelInput
            labelName="Name"
            name="name"
            inputType="text"
            value={this.state.name}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Profile Image"
            name="image"
            inputType="text"
            value={this.state.image}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Age"
            name="age"
            inputType="text"
            value={this.state.age}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Address"
            name="address"
            inputType="text"
            value={this.state.address}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Allergies"
            name="allergies"
            inputType="text"
            value={this.state.allergies}
            changeHandler={this.handleChange}
          />
          <LabelInput
            labelName="Balance"
            name="balance"
            inputType="text"
            value={this.state.balance}
            changeHandler={this.handleChange}
          />
          <label>Parent</label>
          <ParentSelect
            name="parent_id"
            value={this.state.parent_id}
            handleSelectChange={this.handleChange}
          />
          <br />
          <br />
          <input
            name="forms"
            type="file"
            multiple
            onChange={this.handleFilesChange}
          />
          <br />
          <br />
          <button>Create Child</button>
          <CancelButton pathname={this.props.history.location.pathname} />
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    provider: state.provider
  };
};

const mapDispatchToProps = dispatch => {
  return {
    newChild: childInfo => dispatch(createChild(childInfo)),
    setProvider: providerObj => dispatch(setProvider(providerObj))
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(CreateForm)
);
