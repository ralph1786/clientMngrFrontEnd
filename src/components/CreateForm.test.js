import React from "react";
import { CreateForm } from "./CreateForm";
import { shallow } from "enzyme";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<CreateForm history={{ location: {} }} />);
});

it("provide a snapshot of the CreateForm component", () => {
  expect(wrapper).toMatchSnapshot();
});

it("checks that a form is being rendered", () => {
  expect(wrapper.find("form")).toBeTruthy();
});
