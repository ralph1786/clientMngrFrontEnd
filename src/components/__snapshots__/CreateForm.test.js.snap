// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provide a snapshot of the CreateForm component 1`] = `
<div
  className="create-form animated flipInY"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label>
      Image Url
    </label>
    <input
      name="image"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label>
      Age
    </label>
    <input
      name="age"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label>
      Address
    </label>
    <input
      name="address"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label>
      Allergies
    </label>
    <input
      name="allergies"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label>
      Balance
    </label>
    <input
      name="balance"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <label>
      Parent
    </label>
    <withRouter(Connect(ParentSelect))
      handleSelectChange={[Function]}
      name="parent_id"
      value={1}
    />
    <br />
    <br />
    <input
      multiple={true}
      name="forms"
      onChange={[Function]}
      type="file"
    />
    <br />
    <br />
    <button>
      Create Child
    </button>
    <CancelButton />
  </form>
</div>
`;
